plugins/pulp_python/plugins/importers/sync.py

```diff
@@ -206,16 +193,15 @@ def __init__(self, repo, conduit, config, working_dir):
 
     def generate_download_requests(self):
         """
-        For each package that wasn't available locally, yield a Nectar
-        DownloadRequest for its url attribute.
+        Yield a Nectar Download request for each package that wasn't available locally.
 
         :return: A generator that yields DownloadReqests for the Package files.
         :rtype:  generator
         """
-        for p in self.get_local_units_step.units_to_download:
-            url = self.unit_urls.pop(p)
+        for package in self.get_local_units_step.units_to_download:
+            url = os.path.join(self._feed_url, 'packages', package.path)
```

Comment:
It looks like the metadata includes a url field. Even if we don't keep that on the model, we could have a property that creates this url.